account-decoder/src/lib.rs

```diff
@@ -18,6 +18,7 @@ pub mod validator_info;
 
 use {
     crate::parse_account_data::{parse_account_data, AccountAdditionalData, ParsedAccount},
+    base64::{engine::general_purpose::STANDARD as BASE64_STANDARD, Engine as _},
```

Comment:
```suggestion
    base64::{prelude::BASE64_STANDARD, Engine},
```
Seems slightly nicer/cleaner to let base64 do the alias for us.
And I don't think the Engine alias is necessary, is it?

Reply:
> Seems slightly nicer/cleaner to let base64 do the alias for us.

Ah, I missed that.  And I've used the same name as the library author :))

> And I don't think the Engine alias is necessary, is it?

It is necessary, as `encode()` and `decode()` are trait methods now.
Trait needs to be in scope for the methods to be callable.
`base64::prelude` also re-exports `Engine`:
```rust
pub use crate::engine::Engine;
```

Clippy, or even rustc, would complain, if it would not be.

Reply:
See my suggestion above; I'm talking specifically about the alias part: `as _`.
I pulled your branch and rewrote the use statement as just `Engine`, and it compiled fine. Clippy did not complain.

Reply:
> See my suggestion above; I'm talking specifically about the alias part: `as _`. I pulled your branch and rewrote the use statement as just `Engine`, and it compiled fine. Clippy did not complain.

I think I misunderstood what you were suggesting.
In the comment, you said:

> And I don't think the Engine alias is necessary, is it?

And I read it as `Engine import is necessary`, but you do import it in the `use` statement:

```rust
  base64::{prelude::BASE64_STANDARD, Engine},
```

Did not expect you to suggest we import things that we do not need :P
Why import `Engine` if we are not using it?
Adding `as _` seems cleaner, as it shows that we do not need this import, except for the trait implementations it carries.

Reply:
I think this is pretty standard in Rust.
To me, it seems to align with the idea that you do not import things you do not need.
Haskell has a syntax for this case as well, though, as Haskell allows orphan instances, it is probably more common to import just trait implementations from unrelated packages.

Reply:
I would certainly be happy to read any resources you have suggesting this is best practice in Rust, or a clippy lint coming down the pipe; we could consider updating this throughout entire monorepo. In the meantime, I would prefer stick with the style that is the overwhelming precedent in the repo.
The best argument I've seen for this pattern is [when there is a symbol conflict](https://rustwiki.org/en/reference/items/use-declarations.html#underscore-imports), which is the case for one our existing usages and is not the case here with `Engine`.

Reply:
I can not find any articles or research on how common this is.
Probably should not have said "pretty standard" - what I meant is that this is how I've seen it written in the codebases that I worked with.

I am somewhat confused as to why not use a feature that is there.
Rust import style is not to import things that are not needed.

If we need a reference to an authority, a case can be made that Rust mostly copied its type system from Haskell.  Haskell compiler suggests a Haskell equivalent of an `as _`, when emitting warnings for redundant imports.  I can see that the Rust compiler does not include the `as _` in the suggested fix message.

Another case could be the Rust standard library source code.
There are `as _` imports there: https://github.com/search?q=repo%3Arust-lang%2Frust+%2Fuse+.*as+_%2F&type=code
It is not easy for me to count the number of cases when `as _` could have been used but were not.  So I do not know how common or uncommon it is in the standard library.

While I agree that consistency is important, I am not sure if I agree with this argument in this particular case.  As nor the rust compiler, nor Clippy is checking this, code might be written using both styles.  And the reviewer could also ask for `as _` to be added, similarly to how you are asking to remove it :)
I personally think `as _` is a more precise way to write it, so if it is written in a less precise way in most cases in this repo, it could just be an oversight.  I do not think it really contributes to the readability of the codebase.

Anyway, as this PR is not about how imports should be written, I have removed `as _`.